lib/ramble/ramble/cmd/results.py

```diff
@@ -6,25 +6,122 @@
 # option. This file may not be copied, modified, or distributed
 # except according to those terms.
 
+import os
 import json
 
+import spack.util.spack_yaml as syaml
+
 import ramble.experimental.uploader
+import ramble.cmd
+import ramble.reports
 from ramble.util.logger import logger
 
-description = "import experiment results from file"
+description = "take actions on experiment results"
 section = "results"
 level = "short"
 
 
 def setup_parser(subparser):
     sp = subparser.add_subparsers(metavar="SUBCOMMAND", dest="results_command")
 
-    # Upload
     upload_parser = sp.add_parser(
         "upload", help=results_upload.__doc__, description=results_upload.__doc__
     )
     upload_parser.add_argument("filename", help="path of file to upload")
 
+    report_parser = sp.add_parser(
+        "report", help=results_report.__doc__, description=results_report.__doc__
+    )
+    report_parser.add_argument(
+        "--workspace",
+        dest="workspace",
+        metavar="WRKSPC",
+        action="store",
+        help="the workspace to report on",
+    )
+    report_parser.add_argument(
+        "--strong-scaling",
+        dest="strong_scaling",
+        nargs="+",
+        action="append",
+        help="generate a scaling report, requires two args: [performance metric] [scaling metric]"
+        "[optional: group by]",
+        required=False,
+    )
+    report_parser.add_argument(
+        "--weak-scaling",
+        dest="weak_scaling",
+        nargs="+",
+        action="append",
+        help="generate a scaling report, requires two args: [performance metric] [scaling metric]"
+        "[optional: group by]",
+        required=False,
+    )
+    report_parser.add_argument(
+        "--multi-line",
+        dest="multi_line",
+        nargs="+",
+        action="append",
+        help="generate a scaling report, requires two args: [performance metric] [scaling metric]"
+        "[optional: group by]",
+        required=False,
+    )
+    report_parser.add_argument(
```

Comment:
This might be answered as I get through more of the code, but if this is used for filtering how some experiments...
You can use the where common argument (in the common module) for this.

Also, you might want to add  the `--execlude-where` argument to keep the interfaces symmetric.

Reply:
Bob added this and I think it works differently from the common 'where'. It passes in a 'where' string to pandas to downselect the dataframe. It's more like a SQL 'where' statement. I haven't looked into this closely to see if the common 'where' might be useful, but making the pandas query format available adds a lot of control over what shows up in the chart.

Reply:
Will rename to `pandas-where`